scripts/dice-wfrp4e.js

```diff
@@ -43,8 +43,8 @@ class DiceWFRP
       successBonus: 0,
     });
 
-    // Sets/overrides default test diffficulty to Average if Income test
-    sceneStress = testData.income ? "average" : sceneStress;
+    // Sets/overrides default test difficulty to Average if Income or Rest & Recover test
+    sceneStress = testData.income || testData.extra.options.rest ? "average" : sceneStress;
```

Comment:
I don't recall why we didn't do this earlier, if for any reason, but why can this not be 
`sceneStress = dialogOptions.data.testDifficulty || sceneStress`?

I just don't like having to reference income/rest in this function.

Reply:
I think this is actually a requirement to ensure traits get their default difficulty accounted for.

Reply:
Ha! That's hilarious (and reassuring), I was looking again at a way of doing this that took income/rest out of `DiceWFRP` and landed at the same ... 

However, that doesn't work with rest when there's a combat encounter - 
I think because `dialogOptions.data.testDifficulty` for rest is set in `setupCharacteristic` rather than `setupSkill`, so is always undefined going into `DiceWFRP`.  

I may be misdiagnosing that. And as per OP, it's a bit of a nonsense situation to think you can rest with a battle going on around you. But hey ho ... 
- if we're happy that sceneStress overrides rest difficulty, I'm very happy with this option to keep DiceWFRP clean. 
- if not, I guess the rest test needs to be in setupSkill?

(Both approaches seem to work for both trait default difficulty and spells in my testing).

Reply:
I would prefer, just to be consistent, that *anything* that specifies difficulty would override the scene stress default difficulty (whatever the setting, on or off), but I agree it's nonsensical, so I'm not too worried about it.

However I'm a bit confused, what I'm seeing is the opposite of what you're saying. I see the rest option sets the `dialogOptions.data.testDifficulty` in `setupSkill` (line 298), but not in `setupCharacteristic`, which should be after line 214, in exactly the same way as setupSkill does. 

If we can get the default difficult of setupCharacteristic added, we should be able to dereference rest/income in DiceWFRP right?

Reply:
Yep, I'm with you about consistency and dereferencing. Not sure what is going on with what I've seen. Will take another look. Stacked with work next couple of days, but should get back to it at the end of the week.